### PR TITLE
Align behavior of ONNX Frontend operator AveragePool-7, 10, 11, 19 with original framework #20553

### PR DESCRIPTION
**Pull request by Sukhvansh2004**
### Details:
Have created instances of op-7, op-10, op-11 and op-19. 
fixes: https://github.com/openvinotoolkit/openvino/issues/20553#issuecomment-1889780590

### CODE REVIEW

**Comment by Sukhvansh2004** [comment]
image

@gkrivor @p-wysocki  I am getting the following output on running python -m pytest in onnx folder of the repo. Since no test case of average pool is failing I suppose the code is working right?

**Comment by p-wysocki** [comment]
Hi @Sukhvansh2004, the tests are currently passing on master, so your implementation may have somehow broke them, or your ONNX version is not compliant with our requirements.
1. Please make sure that your Python environment has onnx==1.15.0 and run the tests again.
2. If they are still failing, please paste the test errors, we'll take a look.

**Review by mitruska — CHANGES_REQUESTED** [review]
Hello @Sukhvansh2004, thank you for your effort and willingness to make a contribution to the OpenVINO project. 

In general the proposed changes are mixing the OpenVINO Opsets versioning with ONNX Opsets. Looks like OpenVINO `v1::AvgPool` is able to support most of the ONNX AveragePool (7, 10, 11, 19) features that's why the common PoolingFactory is used. 

Definition of the gaps in the existing implementation, can be explored by adding [corresponding tests](https://github.com/openvinotoolkit/openvino/blob/1a8f72f3426e02a2cdd3929b608659cd6003feca/src/frontends/onnx/tests/onnx_import_convpool.in.cpp#L281-L331) for ONNX AvgPool from different ONNX opsets.

Here you can find the disabled tests from the ONNX compliance:
https://github.com/openvinotoolkit/openvino/blob/1a8f72f3426e02a2cdd3929b608659cd6003feca/src/frontends/onnx/tests/tests_python/test_backend.py#L689-L690 https://github.com/openvinotoolkit/openvino/blob/1a8f72f3426e02a2cdd3929b608659cd6003feca/src/frontends/onnx/tests/tests_python/test_backend.py#L742-L747

Most of them are related to the "dilations" attribute, and in fact to cover it, new version of the ov AvgPool is needed (as it was done for ov MaxPool op). But to enable new ov Op not only new class of AvgPool is needed, but also the plugin implementation. 
We will cover such request internally or by a separate, precisely described task for contributors. 

For now I would recommend to add more [tests](https://github.com/openvinotoolkit/openvino/blob/1a8f72f3426e02a2cdd3929b608659cd6003feca/src/frontends/onnx/tests/onnx_import_convpool.in.cpp#L281-L331) and describe, what a single new version of the ov AvgPool should cover. Also feel free to switch to another GFI task.

**Comment by Sukhvansh2004** [comment]
> Hello @Sukhvansh2004, thank you for your effort and willingness to make a contribution to the OpenVINO project.
> 
> In general the proposed changes are mixing the OpenVINO Opsets versioning with ONNX Opsets. Looks like OpenVINO `v1::AvgPool` is able to support most of the ONNX AveragePool (7, 10, 11, 19) features that's why the common PoolingFactory is used.
> 
> Definition of the gaps in the existing implementation, can be explored by adding [corresponding tests](https://github.com/openvinotoolkit/openvino/blob/1a8f72f3426e02a2cdd3929b608659cd6003feca/src/frontends/onnx/tests/onnx_import_convpool.in.cpp#L281-L331) for ONNX AvgPool from different ONNX opsets.
> 
> Here you can find the disabled tests from the ONNX compliance:
> 
> https://github.com/openvinotoolkit/openvino/blob/1a8f72f3426e02a2cdd3929b608659cd6003feca/src/frontends/onnx/tests/tests_python/test_backend.py#L689-L690
> 
> https://github.com/openvinotoolkit/openvino/blob/1a8f72f3426e02a2cdd3929b608659cd6003feca/src/frontends/onnx/tests/tests_python/test_backend.py#L742-L747
> 
> Most of them are related to the "dilations" attribute, and in fact to cover it, new version of the ov AvgPool is needed (as it was done for ov MaxPool op). But to enable new ov Op not only new class of AvgPool is needed, but also the plugin implementation. We will cover such request internally or by a separate, precisely described task for contributors.
> 
> For now I would recommend to add more [tests](https://github.com/openvinotoolkit/openvino/blob/1a8f72f3426e02a2cdd3929b608659cd6003feca/src/frontends/onnx/tests/onnx_import_convpool.in.cpp#L281-L331) and describe, what a single new version of the ov AvgPool should cover. Also feel free to switch to another GFI task.


Hi @mitruska , Thank you so much for your review, since you mentioned that op-7, 10 and 11 are having the saming functionality as op-1  (only a additional attribute of ceil mode and documentation changes were there till op-11 while ceil mode was covered via the m_rounding_type variable) so should I revert the changes of op-7,10 and 11. Also for op-19 as you said dilations would require an addtional plugin so what should I do regarding that.

File: src/core/include/openvino/op/avg_pool.hpp [hunk 1]
```diff
@@ -80,5 +80,299 @@ class OPENVINO_API AvgPool : public Op {
...
+protected:
+    Shape m_kernel;
+    Strides m_strides;
+    Shape m_pads_begin;
+    Shape m_pads_end;
+    bool m_exclude_pad{true};
+    PadType m_auto_pad{PadType::EXPLICIT};
+    op::RoundingType m_rounding_type{op::RoundingType::FLOOR};
+};
+}  // namespace v11
```
[the remark this answers is not in the document]
Comment (Sukhvansh2004) [reply src/core/include/openvino/op/avg_pool.hpp:298]: Ok sir, will try to make the suggested changes in a day or 2.

**Comment by mlukasze** [comment]
hey @Sukhvansh2004 

Are you willing to finish this PR? :)

**Comment by Sukhvansh2004** [comment]
Sorry I wont be able to finish the PR as I have got some other commitments now, however I'll try to find time if I could. Till then you can assign it to anyone else

**Pull request closed without merging**